### PR TITLE
[fix](Nereids): Add varchar literal compare

### PR DESCRIPTION
**Pull request by keanji-x**
Signed-off-by: xiejiann <jianxie0@gmail.com>

# Proposed changes

support "1" = "123"

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.95 seconds
 load time: 651 seconds
 storage size: 17123213298 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230106052418_clickbench_pr_74611.html

**Review by morrySnow — COMMENTED** [review]
add regression test and ut

**Comment by XieJiann** [comment]
>

**Comment by XieJiann** [comment]
> add regression test and ut

done

**Review by morrySnow — DISMISSED** [review]

File: regression-test/suites/nereids_syntax_p0/function.groovy [hunk 1]
```diff
@@ -103,5 +103,10 @@ suite("nereids_function") {
         sql "select convert_to('abc', cast(number as varchar)) from numbers('number'='1')"
         exception "must be a constant"
     }
+
+    test {
+        sql """select "1" == "123", "%%" == "%%" """
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/function.groovy:108]: nit: we need remove syntax `1 == 3`, legacy planner only support `1 = 3`, please fix it in another PR

**Review by morrySnow — APPROVED** [review]

**Pull request merged**